hw/ip/otbn/util/check_call_stack.py

```diff
@@ -28,33 +29,34 @@ def _check_call_stack_insn(insn: Insn, operands: Dict[str, int]) -> bool:
     return True
 
 
-def check_call_stack(program: OTBNProgram) -> Tuple[bool, str]:
+def check_call_stack(program: OTBNProgram) -> CheckResult:
     '''Check that the special register x1 is used safely.
 
     If x1 is used for purposes unrelated to the call stack, it can trigger a
     CALL_STACK error. This check errors if x1 is used for any other instruction
     than `jal` or `jalr`.
     '''
+    out = CheckResult()
     imem_start = program.get_pc_at_symbol('_imem_start')
     imem_end = program.get_pc_at_symbol('_imem_end')
     for pc in range(imem_start, imem_end, 4):
         insn = program.get_insn(pc)
         operands = program.get_operands(pc)
         if not _check_call_stack_insn(insn, operands):
-            return (False, 'check_call_stack: FAIL at PC {:#x}: {} {}'.format(
-                pc, insn.mnemonic, operands))
-    return (True, 'check_call_stack: PASS')
+            out.err(
+                'Potentially dangerous use of the call stack register x1 at PC {:#x}: {}'
+                .format(pc, insn.disassemble(pc, operands)))
```

Comment:
Nit: It might be nice to give the instruction mnemonic in this error message (so a reader can immediately go "eek, I'm using ADD with x1" or whatever)

Reply:
It's actually being printed here: the `insn.disassemble` at the end prints the whole instruction e.g. `addi    x1, x1, 5`